### ISSUE TITLE
Processed templates:  Should never be invalid json encodings.

### ISSUE THREAD

**Issue by jayunit100** [openshift/origin#6903]
Im finding that in 3.1.1.6
```ec2-user@ip-172-18-14-244 openshift_scalability]$ oc version
oc v3.1.1.6
kubernetes v1.1.0-origin-1107-g4c8e6f4```

If I try to create the following invalid template:

```
sudo oc process -f ./content/deploymentConfigTemplate.json
```

is capable of creating a json file which is inconsumable via `oc` itself:

```
{
    "kind": "List",
    "apiVersion": "v1",
    "metadata": {},
    "items": [
        {
            "apiVersion": "v1",
            "data": {
                "ca.crt": "asdf",
                "token": "qweqrer"
            },
            "kind": "Secret",
            "metadata": {
                "labels": {
                    "name": "na"
                }
            },
            "type": "kubernetes.io/service-account-token"
        }
    ]
}
```
TL;DR --- I would say that,  `oc process` should ensure that a template's result is valid json before returning it.  I understand that due to annotations all schemas cant be validated maybe, but at a bare minimum, we should be returning json that is encoded properly or else, if not possible do to a bad template, we should die.

**Comment by jayunit100** [comment]
cc @tiwillia @timothysc @jeremyeder , relevant to openshift SVT team.

**Comment by liggitt** [comment]
The attached snippet is valid JSON...

**Comment by jayunit100** [comment]
*closing* per liggit in IRC , template engine does no schema validation.

**Comment by jayunit100** [comment]
this is really root cause = #6693
